tests/cli/test_convert.py

```diff
@@ -0,0 +1,496 @@
+                'convert_options': {},
+            },
+            ConvertFromFormatSubCommand.parsed_args_to_data_frame_handler_convert_args(
+                vars(self.parser.parse_args([
+                    'convert', 'excel', '-o', 'sheet_name', str(self.input_folder / 'invoice.xlsx')
+                ]))
+            )
+        )
+
+    def test_json_to_ndjson(self):
+        expected_args = {
+            'input-format': 'json',
+            'input': sys.stdin,
+            'json': False,
+            'use_table_schema': None,
+            'create_table_schema': None,
+            **self.default_main_parser_args
+        }
+        with (self.input_folder / 'invoice.json').open() as in_f, self.out_file_path.open('w') as out_f:
+            self.parse_and_execute(['convert', 'json'], expected_args, in_f, out_f)
+        self.compare_json_files(self.out_file_path, self.input_folder / 'invoice.ndjson', is_ndjson_file=True)
+
+    def test_json_to_ndjson_file_path(self):
+        expected_args = {
+            'input-format': 'json',
+            'input': self.input_folder / 'invoice.json',
+            'json': False,
+            'use_table_schema': None,
+            'create_table_schema': None,
+            **self.default_main_parser_args
+        }
+        with self.out_file_path.open('w') as out_f:
+            self.parse_and_execute(
+                ['convert', 'json', f'{self.input_folder}/invoice.json'], expected_args, None, out_f)
+        self.compare_json_files(self.out_file_path, self.input_folder / 'invoice.ndjson', is_ndjson_file=True)
+
+    def test_csv_to_ndjson(self):
+        expected_args = {
+            'input-format': 'csv',
+            'input': sys.stdin,
+            'json': False,
+            'delimiter': ',',
+            'decimal': '.',
+            'use_table_schema': None,
+            'create_table_schema': None,
+            **self.default_main_parser_args
+        }
+
+        with (self.input_folder / 'invoice.csv').open() as in_f, self.out_file_path.open('w') as out_f:
+            self.parse_and_execute(['convert', 'csv'], expected_args, in_f, out_f)
+        self.compare_json_files(self.out_file_path, self.input_folder / 'invoice.ndjson', is_ndjson_file=True)
+
+    def test_csv_to_ndjson_file_path(self):
+        expected_args = {
+            'input-format': 'csv',
+            'input': self.input_folder / 'invoice.csv',
+            'json': False,
+            'delimiter': ',',
+            'decimal': '.',
+            'use_table_schema': None,
+            'create_table_schema': None,
+            **self.default_main_parser_args
+        }
+        with self.out_file_path.open('w') as out_f:
+            self.parse_and_execute(
+                ['convert', 'csv', f'{self.input_folder}/invoice.csv'], expected_args, stub_stdout=out_f)
+        self.compare_json_files(self.out_file_path, self.input_folder / 'invoice.ndjson', is_ndjson_file=True)
+
+    def test_csv_to_json(self):
```

Comment:
When making tests for the same function with different parameters pytest.mark.parametrize can be used (just a tip).

Reply:
Nice!